Observer/SalesOrderSaveAfterObserver.php

```diff
@@ -83,7 +83,8 @@ private function isOrderStatusUpdated(Order $order): bool
 
     private function isOrderPaid(Order $order): bool
     {
-        return $order->getBaseTotalDue() == 0 && $order->getOrigData('base_total_due') != 0;
+        return $order->getBaseTotalDue() == 0 &&
```

Comment:
Since this is a float, I'd recommend doing a check like `$order->getBaseTotalDue() < PHP_FLOAT_EPSILON`. But since this isn't really introduced in this PR, I'll still approve :)

Reply:
Agree, that should go into another Pull Request.